ambari-server/src/main/java/org/apache/ambari/server/controller/internal/ServiceGroupResourceProvider.java

```diff
@@ -389,21 +396,29 @@ protected void deleteServiceGroups(Set<ServiceGroupRequest> request)
 
 
   private void validateCreateRequests(Set<ServiceGroupRequest> requests, Clusters clusters)
-    throws AuthorizationException, AmbariException {
+    throws AuthorizationException, AmbariException, IllegalArgumentException {
 
     AmbariMetaInfo ambariMetaInfo = getManagementController().getAmbariMetaInfo();
     Map<String, Set<String>> serviceGroupNames = new HashMap<>();
     Set<String> duplicates = new HashSet<>();
     for (ServiceGroupRequest request : requests) {
       final String clusterName = request.getClusterName();
       final String serviceGroupName = request.getServiceGroupName();
+      String stackVersion = request.getStackId();
+      //TODO: ******* For test ********* this piece of code should be removed after web UI adds default stack id
+      if (StringUtils.isBlank(stackVersion)) {
+        stackVersion = "HDP-2.0.6";
+        request.setStackId(stackVersion);
+        LOG.info("***** Add fake stack id in order to create cluster from web UI, this code will be removed after web UI adds default stack id");
+      }
```

Comment:
Why is this fake code needed? If the web client is sending down any information about mpacks, then you should be able to use that, right? If no, then just get the first mpack registered for now.

Reply:
Don't hardcode this. Take the value from cluster/version to make it work for single mpack install for now.